v2/header.go

```diff
@@ -115,3 +118,14 @@ func XGoogHeader(keyval ...string) string {
 	}
 	return buf.String()[1:]
 }
+
+// InsertMetadata is for use by the Google Cloud Libraries only.
+//
+// InsertMetadata returns a new metadata.MD with the provided mds merged with
+// any existing metadata in the provided context.
+func InsertMetadata(ctx context.Context, mds ...metadata.MD) metadata.MD {
```

Comment:
Does this need to be variadic? metadata.MD is already a map[string][]string under the hood.

Reply:
Should this be kv pairs instead of metadata.MD?

Reply:
[Usages](https://github.com/googleapis/google-cloud-go/blob/3d1a39907f4569d4014396fb11e26f5b6cb2ea63/accessapproval/apiv1/access_approval_client.go#L548) of the existing generated `InsertMetadata` function in GAPIC clients currently pass two `metadata.MD` arguments to the variadic param. These would need to be merged in the client (adding noise to the client) if this param were changed:

```go
ctx = insertMetadata(ctx, c.xGoogMetadata, md)
```

Reply:
I think the code in the clients can be simplified, instead of storing the base metadata as a metadata.MD we could just store it as a []string and combine them all at once.